Extension/src/LanguageServer/clientCollection.ts

```diff
@@ -116,22 +116,29 @@ export class ClientCollection {
         if (key) {
             this.languageClients.delete(key);
 
+            const newClient: cpptools.Client = this.getClientForFolder(client.RootFolder);
+
             if (transferFileOwnership) {
                 // This will create a new Client for the workspace since we removed the old one from this.languageClients.
                 client.TrackedDocuments.forEach(document => this.transferOwnership(document, client));
                 client.TrackedDocuments.clear();
-            } else {
-                // Create an empty client that will continue to "own" files matching this workspace, but ignore all messages from VS Code.
-                this.languageClients.set(key, cpptools.createNullClient());
```

Comment:
If you remove this, then isn't that going to restart the language server even though we said we wouldn't for this case?